project4/name_server/nameserver.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file Name Server Class
+ */
+#ifndef PROJECT4_NAMESERVER_H
+#define PROJECT4_NAMESERVER_H
+
+#include <consistent_hash_msgs.pb.h>
+
+#include <map>
+#include <string>
+
+namespace nameserver {
+
+/**
+ * @class The basic name server
+ */
+class Nameserver {
+ public:
+  /**
+   * @param the config file of this name server
+   */
+  explicit Nameserver(const std::string& config_file);
+
+  /**
+   * @param Generic NameServerMessage request
+   */
+  void HandleRequest(const consistent_hash_msgs::NameServerMessage &request);
+
+  /**
+   * @param A NameServerMessage request
+   */
+  void HandleRequest(const consistent_hash_msgs::EntranceInformation &request);
+
+  void HandleRequest(const consistent_hash_msgs::ExitInformation &request);
+
+  void HandleRequest(const consistent_hash_msgs::UpdatePredecessorRequest &request);
+
+  void HandleRequest(const consistent_hash_msgs::UpdateSuccessorRequest &request);
+
+  void HandleRequest(const consistent_hash_msgs::UpdatePredecessorResponse &request);
+
+  void HandleRequest(const consistent_hash_msgs::LookUpResult &request);
+
+  void HandleRequest(const consistent_hash_msgs::InsertResult &request);
+
+  void HandleRequest(const consistent_hash_msgs::DeleteResult &request);
+
+  /**
+   * @param predecessor Predecessor if true, else successor
```

Comment:
We could do that, but then it might as well be a static method. I was sorta thinking that this could be a "convenient" method to handle a oneshot requests to a predecessor or successor.

Reply:
I gotcha